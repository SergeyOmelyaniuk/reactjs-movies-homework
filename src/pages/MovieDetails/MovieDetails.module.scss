.movieDetails {
	padding-top: 32px;
	@media (max-width: 1600px) {
		padding-top: 26px;
	}
}

.titleRecommendations {
	font-size: 34px;
	text-transform: uppercase;
	display: block;
	margin-bottom: 50px;
	@media (max-width: 1600px) {
		font-size: 28px;
		margin-bottom: 42px;
	}
}
