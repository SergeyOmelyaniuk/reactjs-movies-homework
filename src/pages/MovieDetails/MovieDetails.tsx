import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import MovieDescription from '../../сomponents/MovieDescription';
import WrapMovies from '../../сomponents/WrapMovies';
import styles from './MovieDetails.module.scss';
import { useAppDispatch, useAppSelector } from '../../store/hooks';
import { fetchMovieDetails } from '../../store/movieSlice';
import { fetchGenres } from '../../store/moviesSlice';
import {
	getListGenres,
	getLanguageSelected,
	getMovieDetails,
	getMovieImages,
	getMoviesRecommendations,
	getTopCast,
} from '../../selectors';
import Preloader from '../../сomponents/Preloader';
import getGenresFilms from '../../helpers/getGenresFilms';
import { translate } from '../../constants';

const numberMoviesPerPage = 5;

const MovieDetails = () => {
	const dispatch = useAppDispatch();
	const { movieId } = useParams<{ movieId: string }>();

	const movieDetails = useAppSelector(getMovieDetails);
	const movieImages = useAppSelector(getMovieImages);
	const moviesRecommendations = useAppSelector(getMoviesRecommendations);
	const topCast = useAppSelector(getTopCast);
	const listGenres = useAppSelector(getListGenres);
	const currentLanguage = useAppSelector(getLanguageSelected);

	useEffect(() => {
		dispatch(fetchGenres());
	}, [dispatch, currentLanguage]);

	useEffect(() => {
		dispatch(fetchMovieDetails(movieId));
		window.scrollTo(0, 0);
	}, [dispatch, movieId, currentLanguage]);

	if (!movieDetails || !movieImages || !moviesRecommendations || !topCast) {
		return <Preloader />;
	}

	const listMoviesRecommendations = moviesRecommendations.slice(
		0,
		numberMoviesPerPage
	);
	const movies = getGenresFilms(listGenres, listMoviesRecommendations);

	return (
		<div className={styles.movieDetails}>
			<MovieDescription
				movieDetails={movieDetails}
				movieImages={movieImages}
				topCast={topCast}
				currentLanguage={currentLanguage}
			/>
			<div className={styles.recommendations}>
				<span className={styles.titleRecommendations}>
					{translate[currentLanguage.title].recommendations}
				</span>
				<WrapMovies movies={movies} />
			</div>
		</div>
	);
};

export default MovieDetails;
