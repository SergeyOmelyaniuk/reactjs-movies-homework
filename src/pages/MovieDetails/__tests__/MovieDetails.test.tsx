import React from 'react';
import { render } from '@testing-library/react';
import MovieDetails from '..';

describe('MovieDetails', () => {
	test('should render', () => {
		const { container } = render(<MovieDetails />);

		expect(container).toMatchSnapshot();
	});
});
