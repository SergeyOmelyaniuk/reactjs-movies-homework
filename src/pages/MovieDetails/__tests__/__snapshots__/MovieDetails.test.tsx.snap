// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should render 1`] = `
<div>
  <div
    class="movieDetails"
  >
    <div
      class="movieDescription"
    >
      <div
        class="wrapPhotoMovie"
      >
        <img
          alt=""
          class="photoMovie"
        />
      </div>
      <div
        class="infoMovie"
      >
        <span
          class="title"
        >
          Title:
        </span>
        <h2
          class="nameMovie"
        >
          Красотка на взводе
        </h2>
        <div
          class="textBlock"
        >
          <span
            class="title"
          >
            Overview:
          </span>
          <p
            class="text"
          >
            Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.
          </p>
        </div>
        <div
          class="textBlock"
        >
          <span
            class="title"
          >
            Release date:
          </span>
          <p
            class="number"
          >
            2021-05-26
          </p>
        </div>
        <div
          class="textBlock"
        >
          <span
            class="title"
          >
            Revenue:
          </span>
          <p
            class="number"
          >
            $120000
          </p>
        </div>
        <div
          class="textBlock"
        >
          <span
            class="title"
          >
            Duration:
          </span>
          <p
            class="number"
          >
            2:14
          </p>
        </div>
        <div
          class="wrapGenre"
        >
          <div
            class="genre"
          >
            Боевик
          </div>
          <div
            class="genre"
          >
            Комедия
          </div>
        </div>
        <div
          class="wrapTopBilledCast"
        >
          <div
            class="topBilledCast"
          >
            <div
              class="header"
            >
              <span
                class="title"
              >
                Top billed cast
              </span>
              <button
                class="button"
              >
                Show all
              </button>
            </div>
            <div
              class="wrap"
            >
              <div
                class="actorCard"
              >
                <div
                  class="wrapImage"
                >
                  <img
                    alt=""
                    class="image"
                  />
                </div>
                <div
                  class="content"
                >
                  <h2
                    class="title"
                  >
                    Kate Beckinsale
                  </h2>
                  <p
                    class="character"
                  >
                    Lindy
                  </p>
                </div>
              </div>
              <div
                class="actorCard"
              >
                <div
                  class="wrapImage"
                >
                  <img
                    alt=""
                    class="image"
                  />
                </div>
                <div
                  class="content"
                >
                  <h2
                    class="title"
                  >
                    Stanley Tucci
                  </h2>
                  <p
                    class="character"
                  >
                    Dr. Ivan Munchin
                  </p>
                </div>
              </div>
              <div
                class="actorCard"
              >
                <div
                  class="wrapImage"
                >
                  <img
                    alt=""
                    class="image"
                  />
                </div>
                <div
                  class="content"
                >
                  <h2
                    class="title"
                  >
                    Bobby Cannavale
                  </h2>
                  <p
                    class="character"
                  >
                    Detective Vicars
                  </p>
                </div>
              </div>
              <div
                class="actorCard"
              >
                <div
                  class="wrapImage"
                >
                  <img
                    alt=""
                    class="image"
                  />
                </div>
                <div
                  class="content"
                >
                  <h2
                    class="title"
                  >
                    Jai Courtney
                  </h2>
                  <p
                    class="character"
                  >
                    Justin
                  </p>
                </div>
              </div>
              <div
                class="actorCard"
              >
                <div
                  class="wrapImage"
                >
                  <img
                    alt=""
                    class="image"
                  />
                </div>
                <div
                  class="content"
                >
                  <h2
                    class="title"
                  >
                    Laverne Cox
                  </h2>
                  <p
                    class="character"
                  >
                    Detective Nevin
                  </p>
                </div>
              </div>
              <div
                class="actorCard"
              >
                <div
                  class="wrapImage"
                >
                  <img
                    alt=""
                    class="image"
                  />
                </div>
                <div
                  class="content"
                >
                  <h2
                    class="title"
                  >
                    David Bradley
                  </h2>
                  <p
                    class="character"
                  >
                    Gareth Fizel
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="blockImages"
        >
          <span
            class="titleImages"
          >
            Images
          </span>
          <div
            class="wrapImages"
          >
            <div
              class="wrapImage"
            >
              <img
                alt=""
                class="image"
              />
            </div>
            <div
              class="wrapImage"
            >
              <img
                alt=""
                class="image"
              />
            </div>
            <div
              class="wrapImage"
            >
              <img
                alt=""
                class="image"
              />
            </div>
            <div
              class="wrapImage"
            >
              <img
                alt=""
                class="image"
              />
            </div>
            <div
              class="wrapImage"
            >
              <img
                alt=""
                class="image"
              />
            </div>
            <div
              class="wrapImage"
            >
              <img
                alt=""
                class="image"
              />
            </div>
            <div
              class="wrapImage"
            >
              <img
                alt=""
                class="image"
              />
            </div>
            <div
              class="wrapImage"
            >
              <img
                alt=""
                class="image"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="recommendations"
    >
      <span
        class="titleRecommendations"
      >
        Recommendations
      </span>
      <div
        class="wrapMovies"
      >
        <div
          class="wrapMovie"
        >
          <div
            class="movieCard"
          >
            <div
              class="wrapImage"
            >
              <div
                class="overlay"
              />
              <img
                alt=""
                class="image"
              />
              <span
                class="rating"
              >
                68
              </span>
              <div
                class="preview"
              />
            </div>
            <div
              class="content"
            >
              <h2
                class="title"
              >
                Бегущий по лезвию 2049
              </h2>
              <p
                class="genre"
              >
                <span>
                  Боевик
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="wrapMovie"
        >
          <div
            class="movieCard"
          >
            <div
              class="wrapImage"
            >
              <div
                class="overlay"
              />
              <img
                alt=""
                class="image"
              />
              <span
                class="rating"
              >
                97
              </span>
              <div
                class="preview"
              />
            </div>
            <div
              class="content"
            >
              <h2
                class="title"
              >
                Клаустрофобы 2: Лига выживших
              </h2>
              <p
                class="genre"
              >
                <span>
                  Боевик
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="wrapMovie"
        >
          <div
            class="movieCard"
          >
            <div
              class="wrapImage"
            >
              <div
                class="overlay"
              />
              <img
                alt=""
                class="image"
              />
              <span
                class="rating"
              >
                89
              </span>
              <div
                class="preview"
              />
            </div>
            <div
              class="content"
            >
              <h2
                class="title"
              >
                В петле времени
              </h2>
              <p
                class="genre"
              >
                <span>
                  Боевик
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="wrapMovie"
        >
          <div
            class="movieCard"
          >
            <div
              class="wrapImage"
            >
              <div
                class="overlay"
              />
              <img
                alt=""
                class="image"
              />
              <span
                class="rating"
              >
                89
              </span>
              <div
                class="preview"
              />
            </div>
            <div
              class="content"
            >
              <h2
                class="title"
              >
                Дальний космос
              </h2>
              <p
                class="genre"
              >
                <span>
                  Боевик
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="wrapMovie"
        >
          <div
            class="movieCard"
          >
            <div
              class="wrapImage"
            >
              <div
                class="overlay"
              />
              <img
                alt=""
                class="image"
              />
              <span
                class="rating"
              >
                55
              </span>
              <div
                class="preview"
              />
            </div>
            <div
              class="content"
            >
              <h2
                class="title"
              >
                Как приручить дракона 3
              </h2>
              <p
                class="genre"
              >
                <span>
                  Боевик
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
